cpp/src/jit/parser.cpp

```diff
@@ -386,53 +324,36 @@ std::string ptx_parser::parse()
   std::string no_comments = remove_comments(ptx);
 
   input_arg_list.clear();
-  // Go directly to the .func mark
-  const size_t length = no_comments.size();
-  size_t start        = no_comments.find(".func") + 5;
-  if (start == length + 5) {
+  auto const _func = std::string(".func");  // Go directly to the .func mark
+  auto f = std::search(no_comments.cbegin(), no_comments.cend(), _func.cbegin(), _func.cend()) +
+           _func.size();
+  if (f >= no_comments.cend()) {
     printf("No function (.func) found in the input ptx code.\n");
     exit(1);
```

Comment:
```suggestion
    CUDF_FAIL("No function (.func) found in the input ptx code.\n");
```
I don't know the existence of `CUDF_FAIL` when I wrote it. : (

Reply:
Done!

Reply:
You can't learn anything unless you don't know something, right? Anyway, this should have come up in code review, so it's not your fault.

The problem is that `CUDF_FAIL` and similar throw exceptions which can be caught at the Cython layer and handled appropriately. `exit` on the other hand will crash the whole process without opportunity to recover.